src/swupd/manifest.go

```diff
@@ -226,3 +226,113 @@ func (m *Manifest) ReadManifestFromFile(f string) error {
 	// return err so the deferred close can modify it
 	return err
 }
+
+// getManifestTimeStamp returns the unix timestamp of m.Header.TimeStamp
+func (m *Manifest) getManifestTimeStamp() int64 {
+	return m.Header.TimeStamp.Unix()
+}
+
+// writeManifestFileHeader writes the header of a manifest to the file
+func writeManifestFileHeader(m *Manifest, w *bufio.Writer) error {
+	var err error
+	if err = m.CheckHeaderPopulated(); err != nil {
+		return err
+	}
+
+	if _, err = w.WriteString(fmt.Sprintf("MANIFEST\t%v\n", m.Header.Format)); err != nil {
+		return err
+	}
+	if _, err = w.WriteString(fmt.Sprintf("version:\t%v\n", m.Header.Version)); err != nil {
+		return err
+	}
+	if _, err = w.WriteString(fmt.Sprintf("previous:\t%v\n", m.Header.Previous)); err != nil {
+		return err
+	}
+	if _, err = w.WriteString(fmt.Sprintf("filecount:\t%v\n", m.Header.FileCount)); err != nil {
+		return err
+	}
+	if _, err = w.WriteString(fmt.Sprintf("timestamp:\t%v\n", m.getManifestTimeStamp())); err != nil {
+		return err
+	}
+	if _, err = w.WriteString(fmt.Sprintf("contentsize:\t%v\n", m.Header.ContentSize)); err != nil {
+		return err
+	}
+
+	for _, include := range m.Header.Includes {
+		if _, err = w.WriteString(fmt.Sprintf("includes:\t%v\n", include.Name)); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// writeManifestFileSep writes a blank line to the file
+func writeManifestFileSep(w *bufio.Writer) error {
+	_, err := w.WriteString("\n")
+	return err
+}
+
+// writeManifestFileEntry writes the file entry line to the file
+func writeManifestFileEntry(file *File, w *bufio.Writer) error {
+	var flags string
+	var err error
+	if flags, err = file.getFlagString(); err != nil {
+		return err
+	}
+
+	line := fmt.Sprintf("%v\t%v\t%v\t%v\n",
+		flags,
+		file.getHashString(),
+		file.Version,
+		file.Name)
+	_, err = w.WriteString(line)
+	return err
+}
+
+// writeManifestFileEntries writes all the entries in files to the file
+func writeManifestFileEntries(m *Manifest, w *bufio.Writer) error {
```

Comment:
Not sure this is worth a separate function. If it is then it should take an *io.Writer.